documentation/modules/cruise-control/proc-generating-rebalance-proposals.adoc

```diff
@@ -0,0 +1,124 @@
+// Module included in the following assemblies:
+//
+// assembly-cruise-control-concepts.adoc
+
+[id='proc-generating-optimization-proposals-{context}']
+= Generating optimization proposals 
+
+When you create or update a `KafkaRebalance` resource, Cruise Control generates an xref:con-optimization-proposals-{context}[optimization proposal] for the Kafka cluster based on the configured xref:con-optimization-goals-{context}[optimization goals].
+
+You can then analyze the summary information in the optimization proposal and decide whether to implement it. 
+
+.Prerequisites
+
+* You have xref:proc-deploying-cruise-control-{context}[deployed Cruise Control] to your {ProductName} cluster.
+
+* You have configured xref:con-optimization-goals-{context}[optimization goals].
+
+.Procedure
+
+. Create a `KafkaRebalance` resource:
+
+.. To use the _default optimization goals_ defined in the `Kafka` resource, leave the `spec` property empty:
++
+[source,yaml,subs="attributes+"]
+----
+apiVersion: {KafkaApiVersion}
+kind: KafkaRebalance
+metadata:
+  name: my-rebalance
+  labels: 
+    strimzi.io/cluster: my-cluster
+spec: {}
+----
+
+.. To set _user-provided optimization goals_ instead of using the default goals, edit the `goals` property and enter one or more goals.
++
+In the following example, rack awareness and replica capacity are user-provided optimization goals:
++
+[source,yaml,subs="attributes+"]
+----
+apiVersion: {KafkaApiVersion}
+kind: KafkaRebalance
+metadata:
+  name: my-rebalance
+  labels: 
+    strimzi.io/cluster: my-cluster
+spec:    
+  goals:
+    - RackAwareGoal,
+    - ReplicaCapacityGoal
+----
+
+. Create or update the resource:
++
+[source,shell,subs="+quotes"]
+----
+kubectl apply -f _your-file_
+----
++
+The Cluster Operator requests the optimization proposal from Cruise Control.
+This might take a few minutes depending on the size of the Kafka cluster. 
+
+. Check the status of the `KafkaRebalance` resource:
++
+[source,shell,subs="+quotes"]
+----
+kubectl get kafka kafkarebalance -o jsonpath='{.status}'
+----
++
+Cruise Control returns one of two statuses:
+
+** `PendingProposal`: The rebalance operator is polling the Cruise Control API to check if the optimization proposal is ready.
+
+** `ProposalReady`: The optimization proposal is ready for review and, if desired, implementation.
+The status contains the optimization proposal as JSON, in the `status.summary` property of the `KafkaRebalance` resource.
+
+. Review the JSON optimization proposal.
```

Comment:
It is not JSON formatted if you use the `kubectl describe` command

Reply:
OK. I've just removed the two mentions that optimization proposals are formatted as JSON.